network/gossip/libp2p/topology/graphsampler.go

```diff
@@ -0,0 +1,17 @@
+package topology
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// GraphSampler is an interface type responsible to sample a connected graph out of a
+// identity list. It is a must-have feature that independent executions of this interface on different
+// nodes (with different seeds) create a connected graph.
+type ConnectedGraphSampler interface {
```

Comment:
Very soon we may not need a `ConnectedGraph` as we will try and go below `n+1/2` fanout. Hence, I feel this interface can be simplified and made to only abstract out how the ids are sampled given a fanout instead of it always trying to return a ConnectedGraph.

Reply:
Actually, the notion of _connected_ sampler is to make sure that whatever implements it should be able to provide a connected component (with some probability). But the aim should be on preserving the connectedness. My aim is to abstract all decisions _affecting_ connectedness of the graph at a single place of the code and let it not be scattered across different modules (as it would be very hard to debug). 

We will use this interface for providing plug-and-play topology constructions with different strategies, e.g., randomized, DHT, etc while keeping the rest of the code unchanged.